supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/MemoryDeclarationValidation.scala

```diff
@@ -0,0 +1,115 @@
+package cromwell.backend.impl.sfs.config
+
+import cromwell.backend.MemorySize
+import cromwell.backend.impl.sfs.config.ConfigConstants._
+import cromwell.backend.validation._
+import wdl4s.expression.NoFunctions
+import wdl4s.parser.MemoryUnit
+import wdl4s.types._
+import wdl4s.values.{WdlFloat, WdlInteger, WdlOptionalValue, WdlValue}
+import wdl4s.{Declaration, NoLookup, WdlExpression}
+
+/**
+  * Maps declarations of memory in WDL runtime attributes to the commands used to submit.
+  *
+  * The wdl runtime attributes for memory specified as strings such as:
+  *
+  * {{{
+  *   runtime {
+  *     memory: "500 MB"
+  *   }
+  * }}}
+  *
+  * However, the backend configuration only supports specifying memory in amounts in Float or Int. To specify the unit
```

Comment:
ToL:

I perhaps could have supported inferred types. In our WDL, runtime attributes are duck typed (this int/float/string looks like a memory value, make it work), while inputs and outputs are strongly typed with implicit conversions (this output is explicitly defined as a WdlFile, but if you give me a WdlString, I'll convert it).

When I implemented the config backend, I went for strongly typed declarations. It was easier, as behind the scenes the config backend is creating a WDL `Task`. The inputs to the task are the backend's `config.runtime-attributes` stanza. From the user's WDL, the `runtime` attributes are passed in to `task.instantiateCommand()`, resulting in a string that may be used to launch the job.

As the generated inputs are WDL declarations, one needs to strongly type them with a WdlType. So `memory: "500 MB"` would need to actually be `String memory: "500 MB"`.

I also needed to know what units to use when passing the memory to the submit string:

```
submit = """docker run --memory-reservation="${memory}m" ${docker} ${script} ..."""
```

In this case, memory should always be in megabytes, even if the wdl says "2GB". I also wanted to know what precision to use, fixed or floating point. So I made it so the person authoring the config could specify the precision and units in one line:

- `Float memory_mb`
- `Int memory_mb`
- `Float memory_gb`
- etc.

With more designing and engineering, perhaps one could change the runtime-attributes configuration from explicitly typed to inferred, more like the WDL `runtime` section:

Instead of defining a backends runtime attributes with:

```
runtime-attributes = """
Float memory_gb = 2.0 # obviously a float
String sge_queue      # mandatory, no default
"""
```

We could use something like:

```
runtime-attributes = """
memory_gb = 2.0  # still obviously a float
sge_queue = null # not sure how to handle mandatory runtime attributes
"""
```

It's possible that something as simple as-- instead of using `Task` inputs, generate a `WdlMap` and pass that to `instantiateCommand`-- might solve the problem of static typing. I believe effectively the `runtime` section of WDL actually is a type of dynamic map also.